### PR TITLE
[Product] Add color scheme picker

### PR DESCRIPTION
**Pull request by ludoboludo**
### PR Summary: 

Add a color picker for the product page

### Why are these changes introduced?

To add more customization possibilities

### What approach did you take?

Added the color picker and applied the class on the top level section element

### Visual impact on existing themes

- Are we wanting to apply a different colour on one of our free theme ? If yes we will need to look at other presets like padding and specifically the margin top on the footer. [Screenshot issue](https://screenshot.click/27-47-tyc6l-qn90l.png)
- Modals like popup block, complementary products with quick add don't get this color scheme applied to them, is that an issue ? [screenshot](https://screenshot.click/27-48-kt6lb-fp7ft.png)
- Let me know where the color picker should be positioned in the sidebar
- Should the product media modal (lightbox zoom when you click on a media) follow the same color scheme treatment ? [gif](https://screenshot.click/27-53-borv7-05g7f.gif)

### Testing steps/scenarios

- [ ] Go to the product page
- [ ] Set the color scheme to something different than background-1
- [ ] Make sure to test all the blocks available for the product info,
- [ ] The different types of media possible (3D, Video and Image)

### Demo links

- [Editor](https://admin.shopify.com/store/os2-demo/themes/157722345494/editor)

### Checklist
- [ ] Added PR summary for [release notes](https://themes.shopify.com/themes/dawn/styles/default#ReleaseNotes)
- [ ] Requested review from UX (Only for changes that are affecting the experience or perceivable visual details)
- [ ] Created a ticket for the [help.shopify.com](https://help.shopify.com) documentation team about updates to theme settings. (Internal-only task)
- [ ] Followed [theme code principles](https://github.com/Shopify/dawn/blob/main/.github/CONTRIBUTING.md#theme-code-principles)
- [ ] Linted with [Theme Check](https://github.com/Shopify/theme-check)
- [ ] Tested on [mobile](https://shopify.dev/themes/store/requirements#mobile-browser-requirements)
- [ ] Tested on [multiple browsers](https://shopify.dev/themes/store/requirements#desktop-browser-requirements)
- [ ] Tested for [accessibility](https://shopify.dev/themes/best-practices/accessibility)

### CODE REVIEW

**Comment by ludoboludo** [comment]
>It is slightly jarring to me as a buyer when I'm browsing products on the collection page and when I click on the quick add modal, I see a [white background](https://screenshot.click/06-58-mwou7-ebkxq-23-2upr7.png), and after clicking View full details for this product, I then see a completely [different background](https://screenshot.click/06-02-pw2h0-3u8ms-23-osnab.png) for the product. For a minute I question whether I clicked on the right product 🤔
What are your thoughts?

It's a tricky one, cause I think the other way around can be true as well 🤔 Being a collection page with quick add and seeing a modal that is a completely different color from the current page you're on can be odd. 
Let's see what UX thinks about it 👍 cc: @danielvan 

Thanks for your feedback and take on those questions @andrewetchen

**Comment by danielvan** [comment]
Thanks for all the questions!

The quick add modal is a tricky one:  if we enable colours, you could have different quick add modal for products that use distinct templates and schemas. That said, the vast majority of users have a single template so I think we should apply the color. My reasoning is that we want more uniformity between quick add and product page, than between collection page and quick add.

For the overlay – great catch: I think it may make sense to apply color there too. If the merchant has picked up a color palette that highlights the product, it would be a shame to lose it on click.

Lastly, on positioning: other sections like Image Banner and Image with text have Schemas before the animation/mobile settings. I think we should do the same.

**Comment by melissaperreault** [comment]
Coming here with fresh eyes here is my take! Let me know if you have concerns! Completing a full review now. 🙏 

> Are we wanting to apply a different colour on one of our free theme ? If yes we will need to look at other presets like padding and specifically the margin top on the footer. [Screenshot issue](https://screenshot.click/27-47-tyc6l-qn90l.png)

No, just adding the flexibility is enough for now.


> Modals like popup block, complementary products with quick add don't get this color scheme applied to them, is that an issue ? [screenshot](https://screenshot.click/27-48-kt6lb-fp7ft.png)

I think Pop up block is fair to keep out, it could get it's own color scheme at some point.
Everything looks good otherwise except missing:
- [ ] [The pickup availability drawer](https://screenshot.click/18-25-sy9w8-g1t37.png) should also be informed by this color scheme. It's only available on the product page and if it gets a color, it would be best to match it instead of relying on a color scheme change on Scheme 1.

> Let me know where the color picker should be positioned in the sidebar

This can't be under `Media` section, it should be above, under the first setting checkbox or get a label to separate it from the Media group. cc. @katycobb What do you think about this hierarchy for the `Product information` section with the new `Color scheme` setting?

Should we also inform where this may apply? At least in the help docs?
e.g. Lightbox, ~Popup modal block~, Quick add modal, Pickup availability drawer

> Should the product media modal (lightbox zoom when you click on a media) follow the same color scheme treatment ?

Yes

**Comment by katycobb** [comment]
Thx for the tag @melissaperreault 

> This can't be under Media section, it should be above, under the first setting checkbox or get a label to separate it from the Media group. cc. @katycobb What do you think about this hierarchy for the Product information section with the new Color scheme setting?

Agreed, it can't be listed under the media heading since it applies to way more than just media. I'd suggest slotting it under the first setting checkbox like Meli suggested. 


>Should we also inform where this may apply? At least in the help docs? e.g. Lightbox, Popup modal block, Quick add modal, Pickup availability drawer

I wouldn't include this as help text in the sidebar since it applies to the vast majority of areas. If there's a place that exists in help docs where we can list exception(s) then +1!

**Comment by ludoboludo** [comment]
Changes made, should be good for a final review 👍 
@eugenekasimov @andrewetchen

**Review by melissaperreault — CHANGES_REQUESTED** [review]
Thanks Ludo! 

- [x] There is an issue with the [Pickup availability container](https://screenshot.click/19-51-7j14i-2qhr7.png)

**Review by eugenekasimov — APPROVED** [review]

**Review by melissaperreault — APPROVED** [review]
Thank you!!! Looks good! 🚢

File: assets/quick-add.js [hunk 1]
```diff
@@ -26,6 +26,11 @@ if (!customElements.get('quick-add-modal')) {
           .then((responseText) => {
             const responseHTML = new DOMParser().parseFromString(responseText, 'text/html');
             this.productElement = responseHTML.querySelector('section[id^="MainProduct-"]');
+            this.productElement.classList.forEach((classApplied) => {
+              console.log(classApplied);
+              if (classApplied.startsWith('color-') || classApplied === 'gradient')
```
Comment (ludoboludo) [inline assets/quick-add.js:31]: @andrewetchen I've added support for the gradient to the modal 👍 
As for the border it was already the case in our current version. If you set background-1 to a gradient, then add som borders on medias. You will see the same result:

**Comment by melissaperreault** [comment]
I am not shocked by the visual impact but we should address it in the release notes:

| Theme | Before | After |
|--------|--------|--------|
| Publisher |  |  |
| Sense |  |  |

**Pull request merged**